babel/messages/pofile.py

```diff
@@ -73,6 +73,151 @@ def denormalize(string):
         return unescape(string)
 
 
+class _PoFileParser(object):
+
+    def __init__(self, locale=None, domain=None, ignore_obsolete=False, charset=None):
+        self.ignore_obsolete = ignore_obsolete
+        self.catalog = Catalog(locale=locale, domain=domain, charset=charset)
```

Comment:
As this is actually just a parser (that adds messages from a po-file style stream into a catalog), it feels like it should be the caller's responsibility to pass in a `catalog` object.

That way this parser could be used as a lower-level API to "append" messages into a premade catalog object.